src/core/thread.d

```diff
@@ -2309,8 +2308,23 @@ private __gshared uint suspendDepth = 0;
  * Throws:
  *  ThreadError if the suspend operation fails for a running thread.
  */
-private void suspend( Thread t ) nothrow
+private bool suspend( Thread t ) nothrow
 {
+    Duration waittime = dur!"usecs"(10);
+ Lagain:
+    // check isRunning in case thread was terminated abnormally
+    if (!t.isRunning)
+    {
+        Thread.remove(t);
+        return false;
+    }
+    else if (atomicLoad(t.m_isInCriticalRegion))
```

Comment:
This is not a change by this PR, but why isn't m_isInCriticalRegion not checked again when the thread is actually suspended. It could have been changed after the check here and the actual suspension.